external-crates/move/move-vm/profiler/src/lib.rs

```diff
@@ -139,18 +139,18 @@ impl GasProfiler {
         frame_display_name: String,
         metadata: String,
     ) -> u64 {
-        *self
-            .shared
-            .frame_table
-            .entry(frame_name.clone())
-            .or_insert({
+        match self.shared.frame_table.get(frame_name.as_str()) {
+            Some(idx) => *idx as u64,
+            None => {
                 let val = self.shared.frames.len() as u64;
                 self.shared.frames.push(FrameName {
                     name: frame_display_name,
                     file: metadata,
                 });
-                val as usize
-            }) as u64
+                self.shared.frame_table.insert(frame_name, val as usize);
+                val
+            }
+        }
```

Comment:
I think you can still use `or_insert_with` here, if you destructure `shared` ahead of time.

```rust
let Shared { frame_table, frames } = &mut self.shared;
frame_table.entry(&frame_name).or_insert_with(|| {
    let val = frames.len();
    frames.push(FrameName {
        name: frame_display_name,
        file: metadata,
    });
    val
}) as u64
```